usaspending_api/broker/management/commands/fabs_nightly_loader.py

```diff
@@ -65,8 +65,44 @@ def delete_stale_fabs(ids_to_delete=None):
         if not ids_to_delete:
             return
 
-        # This cascades deletes for TransactionFABS & Awards in addition to deleting TransactionNormalized records
-        TransactionNormalized.objects.filter(assistance_data__afa_generated_unique__in=ids_to_delete).delete()
+        transactions = TransactionNormalized.objects.filter(
+            assistance_data__afa_generated_unique__in=ids_to_delete).values_list('id')
+        transaction_ids_delete = [str(delete_result[0]) for delete_result in transactions]
+        lt_awards = Award.objects.filter(latest_transaction_id__in=transaction_ids_delete).values_list('id')
+        award_ids_delete = [str(delete_result[0]) for delete_result in lt_awards]
+        lt_trans = TransactionNormalized.objects.filter(award_id__in=award_ids_delete).values_list('id')
+        transaction_ids_delete.extend([str(delete_result[0]) for delete_result in lt_trans])
+        transaction_id_list = ','.join(set(transaction_ids_delete))
+        award_ids_list = ','.join(award_ids_delete)
+
+        db_cursor = connections['default'].cursor()
+
+        # Financial Accounts by Awards
+        fa = 'DELETE ' \
+             'FROM "financial_accounts_by_awards" fa ' \
+             'WHERE fa."award_id" IN ({});'.format(award_ids_list)
+        # Subawards
+        sub = 'DELETE ' \
```

Comment:
Here, i don't think we want to _delete_ the subaward but instead set their `award_id` to null? Since we're still ingesting subawards even if we can't link them to an award

Reply:
Agreed. Slightly worried the script was still doing this for a month now...